src/blackboxTests/src/test/kotlin/org/vaccineimpact/api/blackboxTests/UserTests.kt

```diff
@@ -0,0 +1,56 @@
+package org.vaccineimpact.api.blackboxTests
+
+import com.beust.klaxon.json
+import org.assertj.core.api.Assertions
+import org.junit.Test
+import org.vaccineimpact.api.blackboxTests.helpers.RequestHelper
+import org.vaccineimpact.api.blackboxTests.helpers.TestUserHelper
+import org.vaccineimpact.api.blackboxTests.helpers.validate
+import org.vaccineimpact.api.blackboxTests.schemas.JSONSchema
+import org.vaccineimpact.api.db.JooqContext
+import org.vaccineimpact.api.db.direct.addUserWithRoles
+import org.vaccineimpact.api.models.permissions.ReifiedRole
+import org.vaccineimpact.api.models.Scope
+import org.vaccineimpact.api.models.permissions.PermissionSet
+import org.vaccineimpact.api.test_helpers.DatabaseTest
+
+class UserTests : DatabaseTest()
+{
+    @Test
+    fun `can get user by username`()
+    {
+        validate("/users/testuser") against "User" given {
+            it.addUserWithRoles("testuser", ReifiedRole("member", Scope.Specific("modelling-group", "group")))
```

Comment:
So the issue here is that we want them both to be nullable, but scope_id forms part of the primary key of `user_role`, so it can't be nullable. So we instead use the empty string for that. That's the technical reason in the background.

But what we should return in the API is another matter. I agree that it's kind of nasty. Maybe you could translate the empty string to a null?